mycore-user2/src/main/java/org/mycore/user2/MCRUser.java

```diff
@@ -725,14 +722,15 @@ public void setUserAttribute(String name, String value) {
         if (anyMatch.isPresent()) {
             MCRUserAttribute attr = anyMatch.get();
             attr.setValue(value);
-            getAttributes().removeIf(a -> a.getName().equals(name) && a != attr);
+            getAttributes().removeIf(a -> a.getName().equals(name) && !(a.equals(attr)));
```

Comment:
This would remove all attributes which do not have the same name and value and previously it would remove all attributes which are not the same object. I don't know if this produces any negative side effects.